core/src/utils/framework-delegate.ts

```diff
@@ -89,6 +89,17 @@ export const CoreDelegate = () => {
       BaseComponent.appendChild(el);
 
       await new Promise(resolve => componentOnReady(el, resolve));
+    } else {
+      // If there is no component, then we need to create a new parent
+      // element to apply the css classes to.
+      const el: HTMLElement = BaseComponent.ownerDocument && BaseComponent.ownerDocument.createElement('div');
+      cssClasses.forEach(c => el.classList.add(c));
+      // Move each child from the original template to the new parent element.
+      while (BaseComponent.hasChildNodes()) {
+        el.appendChild(BaseComponent.firstChild);
+      }
+      // Append the new parent element to the original parent element.
+      (BaseComponent as HTMLElement).appendChild(el);
```

Comment:
Another way to do this without the while loop and typecasting:

```suggestion
      const el = BaseComponent.ownerDocument && BaseComponent.ownerDocument.createElement('div');
      cssClasses.forEach(c => el.classList.add(c));
      // Move each child from the original template to the new parent element.
      el.append(...BaseComponent.children);
      
      // Append the new parent element to the original parent element.
      BaseComponent.appendChild(el);
```

Because `el` is of type `Element`, we can use [Element.append()](https://developer.mozilla.org/en-US/docs/Web/API/Element/append) which accepts multiple nodes to insert.

In my (very unscientific) tests I did not see any major performance difference: https://codepen.io/liamdebeasi/pen/GROWRvV

I'll leave it up to you, but this code seems a bit cleaner since it does not require typecasting.

Reply:
I like your approach more. I'll update it!